pkg/controller/component/component_resources.go

```diff
@@ -142,3 +145,66 @@ func newDeploymentConfig(cr *componentsv1alpha1.Component, output *imagev1.Image
 		},
 	}
 }
+
+func getInt32Port(port string) (int32, error) {
+	portNumber, err := strconv.ParseInt(port, 10, 32)
+	if err != nil {
+		return 0, err
+	}
+	return int32(portNumber), nil
+}
+
+func newService(cr *componentsv1alpha1.Component, port string) (*corev1.Service, error) {
+	labels := resource.GetLabelsForCR(cr)
+	int32Port, err := getInt32Port(port)
+	if err != nil {
+		return nil, err
+	}
+	var svcPorts []corev1.ServicePort
+	svcPort := corev1.ServicePort{
+		Name:       cr.Name + "-tcp",
+		Port:       int32Port,
+		Protocol:   corev1.ProtocolTCP,
+		TargetPort: intstr.FromInt(int(int32Port)),
+	}
+	svcPorts = append(svcPorts, svcPort)
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+			Labels:    labels,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: svcPorts,
+			Selector: map[string]string{
+				"deploymentconfig": cr.Name,
+			},
+		},
+	}
+	return svc, nil
+}
+
+func newRoute(cr *componentsv1alpha1.Component) (*routev1.Route, error) {
+	labels := resource.GetLabelsForCR(cr)
+	int32Port, err := getInt32Port(cr.Spec.Port)
+	if err != nil {
+		return nil, err
+	}
+	route := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+			Labels:    labels,
+		},
+		Spec: routev1.RouteSpec{
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: cr.Name,
+			},
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.FromInt(int(int32Port)),
```

Comment:
while trying `oc new-app` I have see when you expose route you see following
```
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  annotations:
    openshift.io/host.generated: "true"
  creationTimestamp: 2019-04-03T11:12:34Z
  labels:
    app: ruby-ex
  name: ruby-ex
  namespace: test
  resourceVersion: "33511"
  selfLink: /apis/route.openshift.io/v1/namespaces/test/routes/ruby-ex
  uid: 6196ac77-5601-11e9-a8ab-525400e16554
spec:
  host: ruby-ex-test.192.168.42.138.nip.io
  port:
    targetPort: 8080-tcp
  to:
    kind: Service
    name: ruby-ex
    weight: 100
  wildcardPolicy: None
```
Just wondering why `TargetPort` does not have `tcp` appended, as we have port name appended with `-tcp` in service. see https://github.com/redhat-developer/devconsole-operator/pull/76/files#diff-3240f219b66b3cb310b8835fe5cf7e42R165

Reply:
425b8f4